exercises/list-ops/test/test_list_ops.c

```diff
@@ -0,0 +1,325 @@
+   check_lists_match(expected, actual);
+
+   delete_list(list);
+   delete_list(expected);
+   delete_list(actual);
+}
+
+static void test_map_non_empty_list(void)
+{
+
+   TEST_IGNORE();
+   list_t *list = new_list(4, (list_value_t[]){ 1, 3, 5, 7 });
+   list_t *expected = new_list(4, (list_value_t[]){ 2, 4, 6, 8 });
+
+   list_t *actual = map_list(list, map_increment);
+   check_lists_match(expected, actual);
+
+   delete_list(list);
+   delete_list(expected);
+   delete_list(actual);
+}
+
+static void test_foldl_empty_list(void)
+{
+   TEST_IGNORE();
+   list_t *list = new_list(0, NULL);
+   list_value_t initial = 2;
+   list_value_t expected = 2;
+
+   list_value_t actual = foldl_list(list, initial, fold_multiply);
+   TEST_ASSERT_EQUAL(expected, actual);
+
+   delete_list(list);
+}
+
+static void test_foldl_direction_independent(void)
+{
+   TEST_IGNORE();
+   list_t *list = new_list(4, (list_value_t[]){ 1, 2, 3, 4 });
+   list_value_t initial = 5;
+   list_value_t expected = 15;
+
+   list_value_t actual = foldl_list(list, initial, fold_add);
+   TEST_ASSERT_EQUAL(expected, actual);
+
+   delete_list(list);
+}
+
+static void test_foldl_direction_dependent(void)
+{
+   TEST_IGNORE();
+   list_t *list = new_list(2, (list_value_t[]){ 2, 5 });
+   list_value_t initial = 5;
+   list_value_t expected = 0;
+
+   list_value_t actual = foldl_list(list, initial, fold_divide);
+   TEST_ASSERT_EQUAL(expected, actual);
+
+   delete_list(list);
+}
+
+static void test_foldr_empty_list(void)
+{
+   TEST_IGNORE();
+   list_t *list = new_list(0, NULL);
+   list_value_t initial = 2;
+   list_value_t expected = 2;
+
+   list_value_t actual = foldr_list(list, initial, fold_multiply);
+   TEST_ASSERT_EQUAL(expected, actual);
+
+   delete_list(list);
+}
+
+static void test_foldr_direction_independent(void)
+{
+   TEST_IGNORE();
+   list_t *list = new_list(4, (list_value_t[]){ 1, 2, 3, 4 });
+   list_value_t initial = 5;
+   list_value_t expected = 15;
+
+   list_value_t actual = foldr_list(list, initial, fold_add);
+   TEST_ASSERT_EQUAL(expected, actual);
+
+   delete_list(list);
+}
+
+static void test_foldr_direction_dependent(void)
+{
+   TEST_IGNORE();
+   list_t *list = new_list(2, (list_value_t[]){ 2, 5 });
+   list_value_t initial = 5;
+   list_value_t expected = 2;
+
+   list_value_t actual = foldr_list(list, initial, fold_divide);
+   TEST_ASSERT_EQUAL(expected, actual);
+
+   delete_list(list);
+}
+
+static void test_reverse_empty_list(void)
+{
+   TEST_IGNORE();
+   list_t *list = new_list(0, NULL);
+   list_t *expected = new_list(0, NULL);
+
+   reverse_list(list);
+   check_lists_match(expected, list);
+
+   delete_list(list);
+   delete_list(expected);
+}
+
+static void test_reverse_non_empty_list(void)
+{
+   TEST_IGNORE();
+   list_t *list = new_list(4, (list_value_t[]){ 1, 3, 5, 7 });
+   list_t *expected = new_list(4, (list_value_t[]){ 7, 5, 3, 1 });
+
+   reverse_list(list);
+   check_lists_match(expected, list);
```

Comment:
The type signature for `reverse_list` suggests the function should return a new `list_t` but this test expects the same list to be reversed and doesn't actually check the return value. This test should probably be:

```c
list_t *actual = reverse_list(list);
check_lists_match(expected, actual);
```

Reply:
I'll just submit a PR